[Doc] Add section about nested resources in `<Breadcrumb>` documentation. (#8648)
marmelab/react-admin: docs/Breadcrumb.md

## `children`

Children of the `<Breadcrumb>` component must be `<Breadcrumb.Item>` components, or any of its derivatives (`<Breadcrumb.ResourceItems>`, `<Breadcrumb.DashboardItem>`). These components can themselves have children to create a breadcrumb path of any depth.

```jsx
import { Breadcrumb } from '@react-admin/ra-navigation';

const MyBreadcrumb = () => (
    <Breadcrumb>
      <Breadcrumb.ResourceItems resources={['otherResources']} />
      <Breadcrumb.Item name="posts" label="Posts">
        <Breadcrumb.Item
          name="edit"
          label={({ record }) => `Edit "${record.title}"`}
          to={({ record }) => record && `${linkToRecord('/songs', record.id)}/edit`}
        />
        <Breadcrumb.Item
          name="show"
          label={({ record }) => record.title}
          to={({ record }) => record && `${linkToRecord('/songs', record.id)}/show`}
        />
        <Breadcrumb.Item name="list" label="My Post List" />
        <Breadcrumb.Item name="create" label="Let's write a Post!" />
      </Breadcrumb.Item>
    </Breadcrumb>
)
```

> See [the `<Breadcrumb.Item>` section](#breadcrumb.item) for more infiormation.

Reviewer: ```suggestion
See [the `<Breadcrumb.Item>` section](#breadcrumbitem) for more information.
```

```
{% endraw %}

## `<Breadcrumb.Item>`

This component renders a single breadcrumb item if the app location matches its `name`.

> ![A breadcrumb item](https://marmelab.com/ra-enterprise/modules/assets/breadcrumb.Item.png)

Reviewer: ```suggestion
![A breadcrumb item](https://marmelab.com/ra-enterprise/modules/assets/breadcrumbItem.png)
```

```jsx
const MyBreadcrumb = () => (
    <Breadcrumb>
        <Breadcrumb.ResourceItems />
        <Breadcrumb.Item name="myhome" label="Home">
            <Breadcrumb.Item name="user" label="User">
                <Breadcrumb.Item name="preferences" label="Preferences" to="/user/preferences" />
            </Breadcrumb.Item>
        </Breadcrumb.Item>
    </Breadcrumb>
);
```

## Nested Resources

When using [nested resources](./Resource.md#nested-resources), create breadcrumb items for the sub-resources. 

> For intance, the screencast at the top of this page shows a `songs` resource nested in an `artists` resources.

Reviewer: ```suggestion
For instance, the screencast at the top of this page shows a `songs` resource nested in an `artists` resources.
```

```jsx
import { Admin, Resource } from 'react-admin';
import { Route } from 'react-router-dom';

export const App = () => (
    <Admin dataProvider={dataProvider}>
        <Resource name="artists" list={ArtistList} edit={ArtistDetail}>
            <Route path=":id/songs" element={<SongList />} />
            <Route path=":id/songs/:songId" element={<SongDetail />} />
        </Resource>
    </Admin>
);
```

This setup creates four routes:

- `/artists` renders the `<ArtistList>` element
- `/artists/:id` renders the `<ArtistDetail>` element
- `/artists/:id/songs` renders the `<SongList>` element
- `/artists/:id/songs/:songId` renders the `<SongDetail>` element

One app location is defined for each route:

- `/artists`: `artists` (defined automatically)
- `/artists/:id`: `artists.edit` (defined automatically)
- `/artists/:id/songs`: `artists.edit.songs` (defined manually)
- `/artists/:id/songs/:songId`: `artists.edit.songs.edit` (defined manually)

Let's see how the components for the songs list and detail pages define their app location:

{% raw %}
```jsx
// in src/songs/SongList.js

> import { useGetOne, List; SearchInput, Datagrid, TextField, DateField } from 'react-admin';

Reviewer: ```suggestion
import { useGetOne, List, SearchInput, Datagrid, TextField, DateField } from 'react-admin';
```

```jsx
// in src/songs/SongDetail.js
import { useGetOne, Edit, SimpleForm, TextInput, DateInput } from 'react-admin';
import { useDefineAppLocation } from '@react-admin/ra-navigation';
import { useParams } from 'react-router-dom';

export const SongDetail = () => {
    const { id, songId } = useParams();
    const { data: record } = useGetOne('artists', { id });
    const { data: song } = useGetOne('songs', { id: songId });
    useDefineAppLocation('artists.edit.songs.edit', { record, song });
    return (
        <Edit resource="songs" id={songId} redirect={`/artists/${id}/songs`}>
            <SimpleForm>
                <TextInput source="title" />
                <DateInput source="released" />
                <TextInput source="writer" />
                <TextInput source="producer" />
                <TextInput source="recordCompany" label="Label" />
            </SimpleForm>
        </Edit>
    );
};
```
{% endraw %}

**Tip**: The `<Edit>` component will call `dataProvider.getOne("songs", { id: songId })` to fetch the song record. Since the `<SongDetail>` component makes the same request, React-admin will deduplicate the calls and only make one request to the dataProvider.

> **Tip**: You don't need to call `useDefineAppLocation` for the pages of the parent resource (artists) as this resource uses the default URLs (`/artists`, `/artists/:id/edit`). The app location will be deduced automatically from these URLs (i.e. `artists` and `artists.edit`).

Reviewer: ```suggestion
**Tip**: You don't need to call `useDefineAppLocation` for the pages of the parent resource (artists) as this resource uses the default URLs (`/artists`, `/artists/:id`). The app location will be deduced automatically from these URLs (i.e. `artists` and `artists.edit`).
```

Careful with that _Copilot_ thing of yours :stuck_out_tongue_closed_eyes:

Author: This one is a pure human error.
